### PR TITLE
AEA-3818 - FHIR API to return more descriptive error code when dosage sequence number missing

### PR DESCRIPTION
**Pull request by natasafrgk**
## Summary

https://nhsd-jira.digital.nhs.uk/browse/AEA-3818

Better Care raised an issue with Live Services because they were receiving a 500 Internal Server Error: "An unknown error occurred processing this request. Contact us for assistance diagnosing this issue" when sending a Create a new prescription - Prescribing request in the INT environment.

On investigation, it was identified that the error condition was generated because the request contained multiple dosage instruction lines but did not include a dosage sequence number, but the error message provided no indication of this.

- Routine Change
 
### Details

A more descriptive message for `Create a new prescription - Prescribing` request is needed when there are more than one `dosageInstruction` items not containing `sequence`  number. In this case a `400` with info about the `missing sequence in multiple dosageInstruction items` needs to be thrown.

Examples were also edited to contain multiple lines.

Additionally, there was a fix as part of tidying up: The eRD example of the same endpoint was failing due to lack of `end` item in `validityPeriod` and this was added as well so we see a `200`.

### CODE REVIEW

File: packages/specification/electronic-prescription-service-api.yaml [hunk 1]
```diff
@@ -388,6 +388,11 @@ paths:
                 description: A valid acute, primary care prescription order request.
                 value:
                   $ref: examples/primary-care/acute/nominated-pharmacy/medical-prescriber/1-Process-Request-Send-200_OK.json
+              acute-multiple-dosages:
+                summary: Acute
```
Comment (kris-szlapa) [inline packages/specification/electronic-prescription-service-api.yaml:392]: Please refrain from using identical names for different examples. 
Please indicate in the name that the example pertains to multiple dosage items.

File: packages/specification/electronic-prescription-service-api.yaml [hunk 2]
```diff
@@ -417,6 +422,11 @@ paths:
                   description: A successful response to an acute, primary care prescription order request.
                   value:
                     $ref: examples/primary-care/acute/nominated-pharmacy/medical-prescriber/1-Process-Response-Send-200_OK.json
+                acute-multiple-dosages:
+                  summary: Acute
```
Comment (kris-szlapa) [inline packages/specification/electronic-prescription-service-api.yaml:426]: Please refrain from using identical names for different examples. 
Please indicate in the name that the example pertains to multiple dosage items.

File: packages/specification/electronic-prescription-service-api.yaml [hunk 3]
```diff
@@ -446,7 +456,7 @@ paths:
                   description: |
                     An error response.
                   value:
-                    $ref: examples/spec/errors/example-a-validation-error-missing-field/Response-FhirError.json
+                    - $ref: examples/spec/errors/example-a-validation-error-missing-field/Response-FhirError.json
```
Comment (kris-szlapa) [inline packages/specification/electronic-prescription-service-api.yaml:459]: I suspect the hyphen was added by mistake.

File: packages/specification/electronic-prescription-service-api.yaml [hunk 4]
```diff
@@ -436,7 +446,7 @@ paths:
             | 400         | MISSING_DIGITAL_SIGNATURE  | Digital signature not found.                                                                                                |
             | 400         | DUPLICATE_PRESCRIPTION_ID  | Duplicate prescription ID exists.                                                                                           |
             | 400         | FAILURE_TO_PROCESS_MESSAGE | Unable to process message for example where the prescription ID has an invalid checksum or an invalid NHS number is passed. |
-
+            | 400         | MISSING_DOSAGE_SEQUENCE    | The request contain multiple dosage instruction lines but no corresponding dosage sequence number                           |
```
Comment (kris-szlapa) [inline packages/specification/electronic-prescription-service-api.yaml:449]: Please arrange the error codes alphabetically in the table.

File: packages/specification/electronic-prescription-service-api.yaml (hunk 3 above)
Comment (natasafrgk) [reply packages/specification/electronic-prescription-service-api.yaml:459]: Correct. I will remove.

File: packages/specification/electronic-prescription-service-api.yaml (hunk 4 above)
Comment (natasafrgk) [reply packages/specification/electronic-prescription-service-api.yaml:449]: I will do. thank you.

File: packages/specification/electronic-prescription-service-api.yaml (hunk 2 above)
Comment (natasafrgk) [reply packages/specification/electronic-prescription-service-api.yaml:426]: This is not identical. It expresses that it contains multiple dosages. I could add more info in the summary but I won't change the title. Thank you.

**Review by kris-szlapa — APPROVED** [review]

**Pull request merged**